src/h3lib/lib/vertex.c

```diff
@@ -0,0 +1,247 @@
+        {{15, 1}, {16, 0}, {-1, 0}, {-1, 0}, {-1, 0}},  // base cell 93
+        {{9, 0}, {13, 3}, {14, 3}, {-1, 0}, {-1, 0}},   // base cell 94
+        {{10, 3}, {15, 0}, {16, 5}, {19, 1}, {-1, 0}},  // base cell 95
+        {{8, 3}, {9, 3}, {13, 0}, {18, 3}, {-1, 0}},    // base cell 96
+        {{13, 3}, {8, 0}, {17, 1}, {12, 4}, {18, 0}},   // base cell 97 (pent)
+        {{8, 3}, {13, 0}, {18, 3}, {-1, 0}, {-1, 0}},   // base cell 98
+        {{16, 1}, {17, 0}, {-1, 0}, {-1, 0}, {-1, 0}},  // base cell 99
+        {{14, 3}, {15, 5}, {19, 0}, {-1, 0}, {-1, 0}},  // base cell 100
+        {{9, 3}, {14, 0}, {19, 3}, {-1, 0}, {-1, 0}},   // base cell 101
+        {{14, 3}, {19, 0}, {-1, 0}, {-1, 0}, {-1, 0}},  // base cell 102
+        {{12, 3}, {17, 0}, {18, 5}, {-1, 0}, {-1, 0}},  // base cell 103
+        {{9, 3}, {13, 0}, {18, 3}, {-1, 0}, {-1, 0}},   // base cell 104
+        {{12, 3}, {16, 1}, {17, 0}, {18, 5}, {-1, 0}},  // base cell 105
+        {{15, 1}, {16, 0}, {17, 5}, {-1, 0}, {-1, 0}},  // base cell 106
+        {{14, 3}, {9, 0}, {18, 1}, {13, 4}, {19, 0}},   // base cell 107 (pent)
+        {{15, 0}, {19, 1}, {-1, 0}, {-1, 0}, {-1, 0}},  // base cell 108
+        {{15, 0}, {16, 5}, {19, 1}, {-1, 0}, {-1, 0}},  // base cell 109
+        {{13, 3}, {18, 0}, {-1, 0}, {-1, 0}, {-1, 0}},  // base cell 110
+        {{13, 3}, {17, 1}, {18, 0}, {-1, 0}, {-1, 0}},  // base cell 111
+        {{14, 3}, {18, 1}, {19, 0}, {-1, 0}, {-1, 0}},  // base cell 112
+        {{16, 1}, {17, 0}, {18, 5}, {-1, 0}, {-1, 0}},  // base cell 113
+        {{14, 3}, {15, 5}, {18, 1}, {19, 0}, {-1, 0}},  // base cell 114
+        {{13, 3}, {18, 0}, {19, 5}, {-1, 0}, {-1, 0}},  // base cell 115
+        {{17, 1}, {18, 0}, {-1, 0}, {-1, 0}, {-1, 0}},  // base cell 116
+        {{15, 5}, {19, 0}, {17, 3}, {18, 2}, {16, 4}},  // base cell 117 (pent)
+        {{15, 5}, {18, 1}, {19, 0}, {-1, 0}, {-1, 0}},  // base cell 118
+        {{13, 3}, {17, 1}, {18, 0}, {19, 5}, {-1, 0}},  // base cell 119
+        {{18, 1}, {19, 0}, {-1, 0}, {-1, 0}, {-1, 0}},  // base cell 120
+        {{17, 1}, {18, 0}, {19, 5}, {-1, 0}, {-1, 0}},  // base cell 121
+};
+
+/**
+ * Get the number of CCW rotations of the cell's vertex numbers
+ * compared to the directional layout of its neighbors.
+ * @return Number of CCW rotations for the cell, or INVALID_ROTATIONS
+ *         if the cell's base cell was not found on this face
+ */
+int vertexRotations(H3Index cell) {
+    // Get the face and other info for the origin
+    FaceIJK fijk;
+    _h3ToFaceIjk(cell, &fijk);
+    int baseCell = H3_EXPORT(h3GetBaseCell)(cell);
+    int cellLeadingDigit = _h3LeadingNonZeroDigit(cell);
+
+    // Excluding the next line from branch coverage because we never reach
+    // the loop ending condition, we always return first
+    for (int i = 0; i < MAX_BASE_CELL_FACES; i++) {  // LCOV_EXCL_BR_LINE
+        BaseCellRotation rotation = baseCellVertexRotations[baseCell][i];
```

Comment:
Out of bounds read here if baseCell > 121. Come to think of it, we have plenty of those when the index is not validated and we index on the base cell.

Reply:
I can validate and return `INVALID_ROTATIONS`.

Reply:
Though actually - `baseCell` isn't from the caller, it's from `h3GetBaseCell`. So assuming a valid H3 input, this should be guaranteed in range.